docs/CONTRIBUTING.md

```diff
@@ -17,22 +10,202 @@ All contributions to this project will be released under the CC0 dedication. By
+
+#### Commenting guidelines
+Commenting is an important part of writing code other people can easily understand (including your future self!). While each developer has a unique style when it comes to writing comments, we have a few guidelines that govern the project, outlined below.
+
+**Function-level comments**
+
+We use [Sphinx](https://www.sphinx-doc.org) to automatically generate html documentation for the PHDI library and rely heavily on function-level documentation to support the automatic generation. The [Sphinx Docstring](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html) documentation describes the basic format that Sphinx requires. The following pieces are expected for all PHDI public functions:
+```
+"""
+Basic description of how the function operates
+
+:param myparam: Parameter descriptions for each function parameter.
+...
+:raises ExceptionType: Exception description for each raised exception.
+"""
+```
+
+The `type` and `rtype` declarations are _not_ used, as they are inferred by Sphinx from the method signature. The `return` declarations are not used either. Any clarifying information about return value shouuld be placed in the basic description.
+
+The [VS Code Extensions](#extensions) section below outlines how the autoDocstring extension can be used to streamline docstring creation in the VS Code IDE.
+
+**Line-level comments**
+
+Commenting brings the need to balance succinctness with verbosity, and clarity with maintainability. Sometimes striking this balance is more of an art than a science. In general, contributors should feel empowered to strike this balance as they see fit. However, in the interest of quality and consistency, we strive for a few principles when adding line-level comments:
+* *Self-documenting code is best*: Use clear, descriptive, and precise names for modules, variables, and functions; write code that speaks for itself.
+* *Consider refactoring*: Modularizing and other restructuring can lead to self-documenting code. Clear, cleanly structured code is better than adding comments to explain poorly written or confusing code.
+* *Comments should add clarity*: Good comments can add information that can't be easily read or inferred, summarize functionality provided by a block of code, or add context to explain non-idiomatic code. Avoid restating information that's already in function-level comments, or can be easily obtained from within the code itself.  
+
+### How to set up your local environment to contribute to our code
+
+#### Hardware
+
+Supported hardware for PHDI tools includes:
+* Linux
+* Windows 10/11
+* Mac (Intel or Apple silicone)
+#### Software
+
+##### Overview
+This document will assume that you're using VSCode as your IDE, but other options (e.g., IntelliJ, Eclipse, PyCharm) can be used as well.
+
+##### Installation
```

Comment:
I think a couple of the following sections can be safely removed - they don't apply to `phdi-azure`, just `phdi`. It's probably safe to remove:
- Installation
- Testing
- All the code coverage sections
- Building the docs